src/main/ts/api.ts

```diff
@@ -19,18 +19,58 @@
  */
 import { parseISO } from 'date-fns';
 import { getJSON } from 'sonar-request';
+import subYears from 'date-fns/subYears';
+import format from 'date-fns/format';
 
-const ROOT = '/api/cayc';
+const MAX_YEARS = 15;
+
+interface Values {
+  val: string, count: number
+}
 
 interface Response {
-  history: Array<{ date: string; value: number }>;
+  facets: Array<{ values: Values[] }>;
 }
 
+const getDateMinusYears = (years: number) => {
+  const date = new Date();
+  date.setFullYear(date.getFullYear() - years);
+  return date;
+};
+
+const makeRequest = (date: Date) =>
+  getJSON('/api/issues/search', {
+    createdAfter: format(subYears(date, 1), 'yyyy-MM-dd'),
+    createdBefore: format(date, 'yyyy-MM-dd'),
+    types: 'BUG,VULNERABILITY',
+    resolved: 'false',
+    facets: 'createdAt'
+  }).then(({ facets }: Response) => facets[0].values);
+
+
+
 export function getIssues() {
-  return getJSON(`${ROOT}/issues_creation_histogram`).then(({ history }: Response) =>
-    history.map(({ date, value }) => ({
-      x: parseISO(date),
-      y: value,
-    }))
-  );
+  const promises = [];
+  for (let i = MAX_YEARS - 1; i >= 0; i--) {
+    const date = getDateMinusYears(i);
+    promises.push(makeRequest(date));
```

Comment:
I just copied the behavior from the Java file, but I wonder if we could make this in a single request.

Reply:
Let's keep it this way, merge for now, and discuss with Zipeng tomorrow. Indeed, it doesn't really make sense to do that, and the ticket description obviously does not mention anything about it...